cmd/insprd/memory/tree/dapp_utils.go

```diff
@@ -179,16 +184,25 @@ func nodeIsEmpty(node meta.Node) bool {
 	return noAnnotations && noName && noParent && noImage
 }
 
-func validBoundaries(appName string, bound meta.AppBoundary, parentChannels map[string]*meta.Channel) string {
-	appBoundary := utils.StringSliceUnion(bound.Input, bound.Output)
-
-	for _, chName := range appBoundary {
-		if parentChannels[chName] == nil {
-			return "invalid app boundary - channel '" + chName + "' doesnt exist in parent app;"
+func validAliases(app *meta.App) (bool, string) {
+	var msg string
+	var valid bool = true
+	for key, val := range app.Spec.Aliases {
+		if ch, ok := app.Spec.Channels[val.Target]; ok {
+			ch.ConnectedAliases = append(ch.ConnectedAliases, key)
+			continue
+		}
+		if app.Spec.Boundary.Input.Union(app.Spec.Boundary.Output).Contains(val.Target) {
+			continue
+		}
+		if valid {
+			valid = false
+			msg = fmt.Sprintf("alias: %s points to an non-existent channel '%s';", key, val.Target)
+		} else {
+			msg = fmt.Sprintf("%s alias: %s points to an non-existent channel '%s';", msg, key, val.Target)
```

Comment:
Is this if really necessary?

One could simply do
```
valid = false
msg = fmt.Sprintf("%s alias: %s points to an non-existent channel '%s';", msg, key, val.Target)
```

The only difference i can see is that the the first caracter of the first error message would have a an empty ' ' blackspace.

That could be fixed by using strings.TrimPrefix(' ')

Reply:
The idea for it is to list all the wrong aliases at once. The suggestion was to store each message in a string slice and, at the end of the function, call a join(";") in it